### PR TITLE
Hero gallery

### PR DESCRIPTION
**Pull request by pelikhan**
This is a little experiment in adding support showing other images, buttons on the hero page. The idea is a simple minimal increment to the existing design that allows us to do campaigns against our own users.: showcase tutorials, game james, etc...

- [x] refactor hero to HeroBanner react comp
- [x] hero cards are loaded from a typically markdown gallery page, specified in pxtarget.json (turns on the feature)
- [x] do some very basic round robin scheme to go through cards
- [x] dot progress ui + click on dot stops carousel
- [x] limit to 5 cards
- [x] auto-massage tutorial link to start in editor

### CODE REVIEW

**Review by eanders-MS — COMMENTED** [review]
Looks good so far. On the design side, I think it's nice to include:
* Dots to indicate how many items total, and the current item.
* Manual forward and backward.
* Pause button.

File: webapp/src/projects.tsx [hunk 1]
```diff
@@ -416,6 +414,98 @@ export class ProjectsMenu extends data.Component<ISettingsProps, {}> {
...
+
+    fetchGallery(): pxt.CodeCard[] {
+        const targetTheme = pxt.appTarget.appTheme;
+        const path = targetTheme.homeScreenHeroGallery;
+        if (!path) {
+            return this.prevGalleries = [];
+        }
+
+        // fetch gallery
+        let res = this.getData(`gallery:${encodeURIComponent(path)}`) as pxt.gallery.Gallery[];
```
Comment (eanders-MS) [inline webapp/src/projects.tsx:475]: Just wondering: how is this a synchronous call?

Comment (pelikhan) [reply webapp/src/projects.tsx:475]: ``gallery`` is a async virtual api; so getData will return undefined until it returns.

**Comment by pelikhan** [comment]
I don't want to add anything fancy until the design meeting decided something. My POV is that this design is well know - e.g. Netflix, Amazon Prime, etc... No need to reinvent the wheel here.

**Comment by pelikhan** [comment]
Adderd dots, pausing and supporting label.

**Comment by pelikhan** [comment]
This PR is ready for review. We can this basic functionality in and refine with @soniakandah 's designs.

**Review by eanders-MS — COMMENTED** [review]
:shipit:

**Pull request merged**